research/03-literate-prog.tex

```diff
@@ -1,6 +1,109 @@
+\newpage
 \section{Literate Programming}
 \label{sec:literate-programming}
 
+Literate programming was first introduced in 1984 for the Pascal
+language~\cite{knuth1984}. Even though the idea was conceived over 30 years
+ago, implementations are not very common. In recent years the idea seems to
+gain popularity again. Examples include the IPython project~\cite{ipython2007},
+published in 2007, as well as Swift Playgrounds~\cite{swift-playgrounds},
+introduced in 2014. 
+Just as with REPLs, the concept of literate programming is known by various
+names in slightly different forms. In this section it is first explained what
+exactly literate programming entails.
```

Comment:
`This section starts with an explanation of what literate programming is exactly`.